src/texttransformation.js

```diff
@@ -121,51 +158,61 @@ export default class TextTransformation extends Plugin {
 		const model = editor.model;
 		const input = editor.plugins.get( 'Input' );
 
-		const configuredTransformations = getConfiguredTransformations( editor.config.get( 'typing.transformations' ) );
+		const testCallback = text => {
+			for ( const transformation of this.configuredTransformations ) {
+				// Get the normalized version of the configured transformation.
+				// See `_setNormalizedTransformations()`, to get more information how key-value pairs are created.
+				const normalizedTransformation = this._normalizedConfiguredTransformations.get( transformation.from );
+				const from = normalizedTransformation.from;
+				const match = from.test( text );
 
-		for ( const transformation of configuredTransformations ) {
-			const from = normalizeFrom( transformation.from );
-			const to = normalizeTo( transformation.to );
+				if ( match ) {
+					return { match, transformation };
+				}
+			}
+		};
 
-			const watcher = new TextWatcher( editor.model, text => from.test( text ) );
+		const watcherCallback = ( evt, data ) => {
+			if ( !input.isInput( data.batch ) ) {
+				return;
+			}
 
-			const watcherCallback = ( evt, data ) => {
-				if ( !input.isInput( data.batch ) ) {
-					return;
-				}
+			const from = normalizeFrom( data.transformation.from );
```

Comment:
Huh, I just realized that we need here only a normalized version, so we can return in this line https://github.com/ckeditor/ckeditor5-typing/pull/223/files#diff-9cd194a3d34419311f88f762b998f2feR170 `normalizedTransformation` and then get rid of the unnecessary normalization here 🤔 Am I right?

Reply:
Done